### PR TITLE
Modifications to form saving trigger and compact PDF

### PR DESCRIPTION
**Pull request by nasirdohwa**
Hello all,

I've made some changes to allow the PDF to be saved when another form is saved. I need this option for some functional requirements and have tested it on my system (v8.9.1). If the same form is selected as the form to be saved, the functionality remains the same. I have also added an option to save PDF as compact version. 

Hopefully you can review and put into main release. 

Cheers,

Nasir.

### CODE REVIEW

**Comment by pbchase** [comment]
@tlstoffs, here's some new code that implements some new features.  You might want to be involved in the review of this to assure the new features are reflected in the documentation.

**Comment by ChemiKyle** [comment]
Hi @nasirdohwa, could you please explain your feature in the `README.md`?

File: ExternalModule.php [hunk 1]
```diff
@@ -25,15 +25,25 @@ function hook_survey_complete($project_id, $record, $instrument, $event_id, $gro
...
       global $Proj, $redcap_version;
 
+      // Get values from settings.
       $source_instruments = AbstractExternalModule::getProjectSetting('ssptf_source_instrument');
+      $active_instruments = AbstractExternalModule::getProjectSetting('ssptf_target_active_field');
+      $compact_pdf = AbstractExternalModule::getProjectSetting('ssptf_compact_pdf');
+      
+      // Safety check to see to make sure compact_pdf has a value as not manditory  
+      if (!$compact_pdf)
+                $compact_pdf = 'false';
```
Comment (jrpence) [inline ExternalModule.php:35]: 'false' is a string.. which will evaluate to true.. https://www.php.net/manual/en/language.types.boolean.php
if your goal is a default of false then set it to _FALSE_

**Comment by pbchase** [comment]
This project is officially end-of-life. The Vanderbilt University Medical Center team incorporated the functionality of this module into REDCap 10.6.0 on 2020-11-30. They are part of the survey option “Save a PDF of completed survey response to a File Upload field”. We recommend all users of the module reconfigure their REDCap projects to use the features in REDCap core. Release 1.1.3 of this module will be the last version released. Users are welcome to fork this module for their own use, but UF CTS-IT will no longer be providing support or updates.

Sorry we never integrated this. :-(

**Pull request closed without merging**